src/main/java/StudentGroup.java

```diff
@@ -0,0 +1,49 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class StudentGroup {
+
+    private Student groupLeader;
+    private final List<Student> students = new ArrayList<>();
+    private final List<String> tasks = new ArrayList<>();
+
```

Comment:
Add here `Map<String, List<Student>> completedTasks = new HashMap<>();` and refactor code to work with this structure to mark tasks for students as completed